File: 2022-2023/sourcecode/lab/lab10/ShoppyJava/app/src/main/java/pl/udu/uwr/pum/shoppyjava/data/ItemViewModel.java
package pl.udu.uwr.pum.shoppyjava.data;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import java.util.List;

public class ItemViewModel extends AndroidViewModel {

    private final ItemRepository repository;

    private final LiveData<List<Item>> readAllData;

    public void insert(Item item) { repository.insert(item); }

    public ItemViewModel(@NonNull Application application) {
        super(application);
        repository = new ItemRepository(application);
        readAllData = repository.ReadAllData();
    }

    public LiveData<List<Item>> getAllData() {
        return readAllData;
    }
}

File: 2022-2023/sourcecode/lab/lab10/ShoppyKotlin/app/src/main/java/pl/udu/uwr/pum/shoppykotlin/viewmodel/ItemViewModel.kt
package pl.udu.uwr.pum.shoppykotlin.viewmodel

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import androidx.lifecycle.viewModelScope
import kotlinx.coroutines.launch
import pl.udu.uwr.pum.shoppykotlin.model.Item
import pl.udu.uwr.pum.shoppykotlin.data.ItemDatabase
import pl.udu.uwr.pum.shoppykotlin.repository.ItemRepository

class ItemViewModel(application: Application) : AndroidViewModel(application) {
    val readAllData: LiveData<List<Item>>
    private val repository: ItemRepository

    init {
        val itemDao = ItemDatabase.getDatabase(application).itemDao()
        repository = ItemRepository(itemDao)
        readAllData = repository.readAllData
    }

    fun addItem(item: Item){
        viewModelScope.launch {
            repository.addItem(item)
        }
    }

    fun getItem(id: Int): LiveData<Item>{
        return repository.getItem(id)
    }

    fun updateItem(item: Item){
        viewModelScope.launch {
            repository.updateItem(item)
        }
    }
}